src/application-error-boundary/ApplicationErrorBoundary.jsx

```diff
@@ -76,14 +83,18 @@ class ApplicationErrorBoundary extends React.Component {
   }
 
   render() {
-    const { children } = this.props;
+    const { children, intl } = this.props;
     const activeError = this.state.error || this.errorRef.current;
 
     if (activeError) {
+      let errorText = activeError.toString();
+      if (!activeError.message || activeError.message.length === 0) {
+        errorText = intl.formatMessage({ id: 'terraApplication.errorBoundary.defaultErrorMessage' });
+      }
       return (
         <StatusView
           variant="error"
-          message={activeError.toString()}
+          message={errorText}
```

Comment:
Could I get some more information about why we're changing the error boundary text? What cases are they being used for? I assumed they would be here in the case of major app failures. Are people seeing these a lot?

Reply:
If there's an error thrown without any error message, then the error view will display the error type without any indication to the user about what went wrong. This would provide some sort of information to the user in that case. This would likely only happen if a custom error was thrown  that didn't provide any message and was not caught anywhere.

Reply:
Have we seen instances where the thrown js exception doesn't have an error message? I don't mean to be difficult but I wondering what brought this change up?

Reply:
@mjhenkes, UX didn't like it.  Didn't think it was user friendly